modules/rust_examples/RsExample/src/eventually_generated.rs

```diff
@@ -16,6 +16,71 @@ use std::collections::HashMap;
 use std::pin::Pin;
 use std::sync::Arc;
 
+/// Struct holding the config for the `foobar` interface.
+#[derive(Debug)]
+pub struct FoobarConfig {
+    /// An interface level bool config.
+    pub some_bool_config: bool,
+
+    /// An interface level integer config.
+    pub some_integer_config: i64,
+}
+
+/// Struct holding the module level config.
+#[derive(Debug)]
+pub struct ModuleConfig {
+    /// A module level string config.
+    pub some_string_config: String,
+
+    /// A module level number config   
+    pub some_number_config: f64,
+
+    /// The config for the `foobar` interface.
+    pub foobar_config: FoobarConfig,
+}
+
+/// Returns the config for the whole module. You can call this function at any
+/// time in your code.
+pub fn get_config() -> ModuleConfig {
+    let raw_config = everestrs::get_module_configs();
+    let foobar_config = FoobarConfig {
+        some_bool_config: raw_config
```

Comment:
I think `raw_config["foobar"]["get_some_bool_config"].try_into().unwrap()` should work and is equivalent.

however, since you are always anyways just `try_into().unwrap()`, I think you should instead in the other one just implement Into and panic there. That makes the code simpler and gets rid of an error that youl anyways never want the user to see.

Reply:
Its a little bit a question how we're perceiving the framework and code libraries. Since for now the framework is stand alone we should not panic inside the framework just because the consumer can't handle errors. If we will change this mind set in the future, we can reflect it in the interfaces